Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java

```diff
@@ -448,6 +459,50 @@ void testGetExamById() {
         assertEquals(exam, actualExam);
     }
 
+    @Test
+    void GivenExamWithInvisibleProblem_WhenGetProblem_ShouldSucceed() throws Exception {
+        ExamView exam = createExamAndGet(now(), now(), "sample-exam");
+        createQuestion(exam.id, PROBLEM_ID);
+        addExaminees(exam.id, STUDENT_A_EMAIL);
+
+        ProblemView exceptedProblem = getBody(getExamProblem(exam.id, STUDENT_A_ID, PROBLEM_ID)
+                .andExpect(status().isOk()), ProblemView.class);
+
+        assertEquals(problem, exceptedProblem);
+    }
+
+    @Test
+    void GivenExamWithVisibleProblem_WhenGetProblem_ShouldSucceed() throws Exception {
```

Comment:
Do we need to test getting a visible problem?
Because getting a visible problem always succeeds.

Reply:
"visible problem always succeeds" is the contract in `getProblem` API in the ProblemService.
We don't have this contract nor context in the ExamService, so I prefer to keep this testcase.

Reply:
The ```ProblemServiceDriver``` in your usecase will use admin token.